pootle/apps/pootle_store/views.py

```diff
@@ -437,6 +333,181 @@ def source_language(self):
+        "submitter__username", "submitter__email", "submitter__full_name",
+        "suggestion_id", "suggestion__target_f", "suggestion__user__full_name",
+        "suggestion__user__username", "quality_check__name"]
+
+    _language_code = None
+
+    @cached_property
+    def permission_context(self):
+        self.object = self.get_object()
+        return self.project.directory
+
+    @property
+    def project(self):
+        return self.object.store.translation_project.project
+
+    def get_queryset(self):
+        return Unit.objects.get_translatable(self.request.user).select_related(
+            "store__parent",
+            "store__translation_project__project",
+            "store__translation_project__language",
+            "store__translation_project__directory")
+
+    def _get_suggestion_description(self, submission):
+        user_url = reverse(
+            'pootle-user-profile',
+            args=[submission["suggestion__user__username"]])
+        display_name = (
+            submission["suggestion__user__full_name"].strip()
+            if submission["suggestion__user__full_name"].strip()
+            else submission["suggestion__user__username"])
+        author = format_html(u'<a href="{}">{}</a>', user_url, display_name)
+        return {
+            SubmissionTypes.SUGG_ADD: _(u'Added suggestion'),
+            SubmissionTypes.SUGG_ACCEPT: _(u'Accepted suggestion from %s',
+                                           author),
+            SubmissionTypes.SUGG_REJECT: _(u'Rejected suggestion from %s',
+                                           author)}.get(submission['type'], None)
+
+    @property
+    def timeline(self):
+        submission_filter = (
+            Q(field__in=[SubmissionFields.TARGET, SubmissionFields.STATE,
+                         SubmissionFields.COMMENT, SubmissionFields.NONE])
+            | Q(type__in=SubmissionTypes.SUGGESTION_TYPES))
+        return (
+            Submission.objects.filter(unit=self.object)
+                              .filter(submission_filter)
+                              .exclude(field=SubmissionFields.COMMENT,
+                                       creation_time=self.object.commented_on)
+                              .order_by("id"))
+
+    @property
+    def language(self):
+        return self.object.store.translation_project.language
+
+    def add_creation_entry(self, grouped_entries):
+        User = get_user_model()
+        has_creation_entry = (
+            len(grouped_entries) > 0
+            and grouped_entries[0]['datetime'] == self.object.creation_time)
+        if (has_creation_entry):
+            grouped_entries[0]['created'] = True
+        else:
+            created = {
+                'created': True,
+                'submitter': User.objects.get_system_user()}
+            if self.object.creation_time:
+                created['datetime'] = self.object.creation_time
+            grouped_entries[:0] = [created]
+        return grouped_entries
```

Comment:
All this piece of code looks like it comes from the possible absence of a creation submission. My guess is that it would be cleaner to add a migration that create those missing submissions so we can simplify this bit of code to something like:

```python
    def add_creation_entry(self, grouped_entries):
        grouped_entries[0]['created'] = True  # There is always some submission.
        return grouped_entries
```

But I am fine with keeping this as is and get the PR landed as soon as possible. This improvement can be seen as some further future post-landing cleanup.